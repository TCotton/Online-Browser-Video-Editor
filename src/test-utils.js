import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import counterReducer from './components/slices/counterSlice';
import filesReducer from './components/slices/filesSlice';
import videoReducer from './components/slices/videoSlice';
import playerReducer from './components/slices/playerSlice';
import audioReducer from './components/slices/audioSlice';
import imageReducer from './components/slices/imageSlice';
import waveformReducer from './components/slices/waveformSlice';

function render(
    ui,
    {
        initialState,
        store = createStore(combineReducers({
            counter: counterReducer,
            files:  filesReducer,
            video: videoReducer,
            player: playerReducer,
            audio: audioReducer,
            images: imageReducer,
            waveform: waveformReducer,
        }), initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react'
// override render method
export { render }