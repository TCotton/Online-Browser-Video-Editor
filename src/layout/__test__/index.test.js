import React from 'react'
import MainLayout from '../index'
import { render } from '../../test-utils'

describe('MainLayout', () => {
  it('should render successfully', () => {
    const { container } = render(<MainLayout />)
    expect(container.querySelector('[data-testid="wrapper"]')).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<MainLayout />)
    expect(baseElement).toMatchSnapshot()
  })
})
