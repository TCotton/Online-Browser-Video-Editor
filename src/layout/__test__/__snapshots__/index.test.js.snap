// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should match snapshot 1`] = `
<body>
  <div>
    <div
      class="wrapper"
      data-testid="wrapper"
    >
      <main>
        <header>
          <img
            alt=""
            src="./had.svg"
          />
        </header>
        <article>
          <section
            class="top"
          >
            <div
              class="column-one"
            >
              <div
                class="filelists"
                data-testid="FileListComponent"
              >
                <section
                  class="file-row"
                  data-testid="FileListComponents"
                />
              </div>
              <section
                class="message"
                data-testid="message"
              >
                <p>
                  Upload and transcode video files for use on social media.
                </p>
                <p>
                  This is a lockdown project by 
                  <a
                    href="https://andywalpole.me/blog/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Andrew Walpole
                  </a>
                  , built using D3.js, the Web Audio API, IndexedDB and WebAssembly (WASM) on a Gatsby/React/Redux base. 
                  <a
                    class="demo"
                    data-testid="link"
                    href="/"
                  >
                    Click here to load a demo video file
                  </a>
                  . 
                  <a
                    href="https://github.com/TCotton/Online-Browser-Video-Editor"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    This is an alpha release
                  </a>
                  .
                </p>
              </section>
            </div>
            <div
              class="column-two"
            >
              <div
                class="viewer"
              />
            </div>
            <div
              class="column-three"
            >
              <div
                class="levels"
              >
                <div
                  class="left default"
                  data-testid="ChannelDisplay"
                >
                  <svg
                    data-testid="svg"
                    height="100%"
                    preserveAspectRatio="none"
                    viewBox="0 0 250 250"
                    width="100%"
                  >
                    <g
                      class="rects"
                      data-testid="rects"
                      transform="translate(0,0)"
                    >
                      <rect
                        class="rect"
                        fill="rgb(32, 224, 187)"
                        height="0"
                        width="250"
                        y="0"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  class="right default"
                >
                  <svg
                    data-testid="svg"
                    height="100%"
                    preserveAspectRatio="none"
                    viewBox="0 0 250 250"
                    width="100%"
                  >
                    <g
                      class="rects"
                      data-testid="rects"
                      transform="translate(0,0)"
                    >
                      <rect
                        class="rect"
                        fill="rgb(14, 175, 163)"
                        height="0"
                        width="250"
                        y="0"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </div>
          </section>
          <section
            class="middle"
          >
            <div
              class="column-one"
            >
              <div
                class="time"
              >
                <div
                  class="admin-icon"
                >
                  <img
                    alt="admin"
                    class="admin"
                    src="./admin.svg"
                  />
                  <time
                    class="duration"
                    data-testid="duration"
                  >
                    0:00.00
                  </time>
                    
                  <span
                    class="break"
                  >
                    |
                  </span>
                    
                  <time
                    data-testid="time"
                  >
                    0:00.00
                  </time>
                </div>
              </div>
            </div>
            <div
              class="column-two"
            >
              <ul
                data-testid="interface"
              >
                <li>
                  <button>
                    <img
                      alt="right cut"
                      class="right-cut"
                      src="./cut.svg"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-describedby="tooltip3"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="backward"
                      class="backward"
                      data-testid="reverse"
                      src="./back.svg"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-describedby="tooltip8"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="stop"
                      class="stop"
                      data-testid="pause"
                      src="./pause.svg"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-describedby="tooltip4"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="play"
                      class="play"
                      data-testid="play"
                      src="./play.svg"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-describedby="tooltip5"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="forward"
                      class="forward"
                      data-testid="forward"
                      src="./forward.svg"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-describedby="tooltip9"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="sound"
                      class="sound"
                      data-testid="sound"
                      src="./sound.svg"
                    />
                  </button>
                </li>
                <li>
                  <div
                    aria-describedby="tooltip6"
                    tabindex="0"
                  >
                    <div
                      class="app"
                    >
                      <form
                        action="#"
                        data-testid="form"
                      >
                        <div
                          class="fileStyling"
                        >
                          <input
                            accept="video/*"
                            data-testid="file"
                            id="file"
                            name="file"
                            required=""
                            type="file"
                            value=""
                          />
                          <label
                            for="file"
                          >
                            Choose a file
                          </label>
                        </div>
                        <div
                          class="Errors"
                        />
                      </form>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    aria-describedby="tooltip7"
                    tabindex="0"
                  >
                    <button
                      aria-controls="FileExport"
                      aria-expanded="false"
                      aria-haspopup="dialog"
                      type="button"
                    >
                      <img
                        alt="export"
                        class="export"
                        data-testid="file-export"
                        src="./export.svg"
                      />
                    </button>
                  </div>
                </li>
              </ul>
            </div>
          </section>
          <section
            class="bottom"
            data-testid="TimeLines"
          >
            <div
              class="row-timeline"
            >
              <div
                class="controls"
              />
              <div
                class="timeline"
              >
                <div
                  class="timeline-ticks"
                >
                  <svg
                    data-testid="svg"
                    preserveAspectRatio="xMidYMid slice"
                    viewBox="0 0 807 30"
                  >
                    <g
                      class="axis-top"
                      data-testid="axis"
                      fill="none"
                      font-family="sans-serif"
                      font-size="10"
                      text-anchor="middle"
                      transform="translate(0,0)"
                    >
                      <path
                        class="domain"
                        d="M0.5,-6V0.5H807.5V-6"
                        stroke="currentColor"
                      />
                      <g
                        class="tick"
                        opacity="1"
                        transform="translate(404,0)"
                      >
                        <line
                          stroke="currentColor"
                          y2="-6"
                        />
                        <text
                          dy="0em"
                          fill="currentColor"
                          style="text-anchor: end; font-size: 1.75ex; font-family: monospace; fill: #ffffff;"
                          transform="translate(10,30) rotate(45)"
                          y="-9"
                        />
                      </g>
                    </g>
                    <g
                      class="waveform"
                      data-testid="waveform"
                    >
                      <path
                        class="svg-line"
                        d="M403.5,15Z"
                        fill="#ffffff"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="row-one"
            >
              <div
                class="controls"
              >
                <button>
                  <img
                    alt=""
                    class="eye"
                    src="./eye.svg"
                  />
                </button>
              </div>
              <div
                class="timeline"
              />
            </div>
            <div
              class="row-two"
            >
              <div
                class="controls"
              >
                <button>
                  <img
                    alt=""
                    class="eye"
                    src="./eye.svg"
                  />
                </button>
              </div>
              <div
                class="timeline"
              >
                <svg
                  data-testid="svg"
                  height="98"
                  preserveAspectRatio="xMidYMid slice"
                  viewBox="0 0 807 98"
                  width="807"
                >
                  <g
                    class="area"
                    data-testid="area"
                    transform="translate(0,0)"
                  >
                    <path
                      class="wave"
                      d="M403.5,49L403.5,49Z"
                      transform="translate(0,0)"
                    />
                  </g>
                </svg>
              </div>
            </div>
            <div
              class="row-three"
            >
              <div
                class="controls"
              />
              <div
                class="timeline"
              >
                <div
                  class="timeline-display"
                  data-testid="TimeDisplay"
                >
                  <test-file-stub
                    style="transform: translate(-17px, -230px);"
                  />
                </div>
              </div>
            </div>
          </section>
        </article>
      </main>
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip3"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Backward
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip8"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Stop
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip4"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Play
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip5"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Forward
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip9"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Mute/Sound
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip6"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Upload
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      class="css-jhczzu"
      data-dialog-ref="FileExport"
      hidden=""
      style="display: none;"
    >
      <div
        aria-label="Welcome"
        aria-modal="true"
        class="css-jkbxbt"
        data-dialog="true"
        hidden=""
        id="FileExport"
        role="dialog"
        style="display: none;"
        tabindex="-1"
      >
        <div>
          <div
            class="export"
            data-testid="export"
          >
            <header>
              <p
                class="title"
              >
                Click below to export
              </p>
              <p
                class="status"
              />
            </header>
            <ul
              class="social"
            >
              <li>
                <button
                  data-testid="twitter"
                >
                  <img
                    alt="click to export for twitter"
                    height="400"
                    src="./social/twitter.svg"
                    width="400"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="facebook"
                >
                  <img
                    alt="click to export for Facebook"
                    height="250"
                    src="./social/facebook.png"
                    width="250"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="linkedin"
                >
                  <img
                    alt="click to export for Linkedin"
                    height="540"
                    src="./social/linkedin.png"
                    width="650"
                  />
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip7"
      role="tooltip"
      style="display: none; position: fixed; left: 0px; top: 0px; pointer-events: none;"
    >
      Export
    </div>
  </div>
</body>
`;
